demo/mkldnn_quant/README_en.md

```diff
@@ -0,0 +1,181 @@
+
+**The parameters during quant-aware training:**
+- **quantize_op_types:** In PaddlePaddle, CPU supports quantizable ops: `depthwise_conv2d`, `conv2d`, `fc`, `matmul`, `transpose2`, `reshape2`, `pool2d`, `scale`, `concat`. However, when inserting fake quantize/dequantize op during training, you only need to insert fake quantize/dequantize ops before and after the first four ops, because the latter five ops: `transpose2`, `reshape2`, `pool2d`, `scale`, `concat`, their input and output scales can be obtained from the scales of other ops before and after, so setting `quantize_op_types` parameter `depthwise_conv2d`, `conv2d`, `fc`, `matmul` is enough.
+- **Other parameters:** Please refer to [PaddleSlim quant_aware API](https://paddlepaddle.github.io/PaddleSlim/api/quantization_api/#quant_aware)
+
+#### 2.2 Post-training quantization
+
+To generate post-training fake quantized model, refer to [Offline post-training quantization tutorial](https://paddlepaddle.github.io/PaddleSlim/tutorials/quant_post_demo/#_1)
+
+## 3. Convert the fake quantized model to DNNL INT8 model
+In order to deploy on the CPU, we need to collect scales, remove the fake quantize/dequantize op, fuse some ops, and finally convert fake quantized ops to real INT8 ops (values in INT8 range, and datatype is int8 type too). Run the following script you can convert fake quantized model to DNNL INT8 model. The script locates [save_quant_model.py](https://github.com/PaddlePaddle/Paddle/blob/develop/python/paddle/fluid/contrib/slim/tests/save_quant_model.py). Copy the script to the directory where the demo is located: `/PATH_TO_PaddleSlim/demo/mkldnn_quant/quant_aware/` and run it as follows:
+```
+python save_quant_model.py --quant_model_path=/PATH/TO/SAVE/FLOAT32/quant/MODEL --int8_model_save_path=/PATH/TO/SAVE/INT8/MODEL
+```
+
+**Available options in the above command and their descriptions are as follows:**
+- **quant_model_path:** input model path, required. A quant model for quantifying training output.
+- **int8_model_save_path:** The final INT8 model output path after the quant model is optimized and quantized by DNNL.
+- **ops_to_quantize:** A comma separated list of specified op types to be quantified. Optional, default is empty, empty means to quantify all quantiable ops. At present, for the image classification and natural language processing models listed in Benchmark, quantizing all quantiable ops can obtain the best accuracy and performance, so it is recommended that users not set this parameter.
+- **--op_ids_to_skip:** A list of op id numbers separated by commas, optional. Default value is none. The op numbers in this list will not be quantified and adopt the FP32 type. To get the ID of a specific op, first run the script using the `--debug` option, and open the generated file `qat_int8_cpu_quantize_placement_pass.dot` to find the op that does not need to be quantified, and the ID number is in parentheses after the Op name.
```

Comment:
"A comma-separated list of operator ID numbers. Optional. Default value: none. Operators with ID in the list will be excluded from quantization and kept as FP32 type. In order to get the ID of a particular operator in the graph, run the script with the `--debug` option, then open the generated file `int8_<number>_cpu_quantize_placement_pass.dot` and find the operator node of your interest. The ID of the operator is shown in parentheses after the operator's name in the node."